lsy_drone_racing/envs/real_race_env.py

```diff
@@ -289,7 +284,73 @@ def send_action(self, action: NDArray):
             # Make sure to use the legacy estimator with the state interface
 
-    def _connect_to_drone(self, radio_id: int, radio_channel: int, drone_id: int) -> Crazyflie:
+    def _prepare_track(self, deploy_config: ConfigDict, track: ConfigDict):
+        assert (not track.get("randomize", False)) or (track.get("randomize", False) and deploy_config.get("real_track_objects", True)), "With a randomly generated track, the deploy.real_track_objects must be set to true."
+
+        if deploy_config.get("real_track_objects", True):
+            self.gates.pos, self.gates.quat, self.obstacles.pos = self._query_track_obj_pose()
+            if track.randomize:
+                self.gates.nominal_pos, self.gates.nominal_quat, self.obstacles.nominal_pos = randomize_track(
+                    gates_pos=self.gates.pos,
+                    gates_quat=self.gates.quat,
+                    obstacles_pos=self.obstacles.pos,
+                    rng_config=self.randomizations
+                )
+                # TODO: The gate safety limits are now hardcoded: 0.5 meters inside the drone safety limits.
+                gate_safety_limits = ConfigDict(
+                    {
+                        'pos_limit_low' : np.array(track.safety_limits.pos_limit_low[:2]) + 0.5,
+                        'pos_limit_high' : np.array(track.safety_limits.pos_limit_high[:2]) - 0.5
+                    }
+                )
+                check_gates_layout(
+                    gates=self.gates,
+                    limits=gate_safety_limits
+                )
+
+            # With generated track, the track objects are by nature in the randomization tolerance —— no need to check track!
+            else:
+                if deploy_config.get("check_race_track", True):
+                    check_race_track(
+                        gates=self.gates,
+                        obstacles=self.obstacles,
+                        rng_config=self.randomizations,
+                )
+    
+    def _prepare_drones(self, deploy_config: ConfigDict, track: ConfigDict):
+        if track.randomize:
+            # The starting position drones.pos has must be set here even though they are not used in _step(). Since they are the target pose for return controller.
+            drone_pos, drone_quat = self._query_drone_pose()
+            drone_pos[2] = 0.01  # Set z to ground level
+            drone_rpy = R.from_quat(drone_quat).as_euler("xyz", degrees=False)
+            self.drones.pos[self.rank], self.drones.rpy[self.rank] = drone_pos, drone_rpy
+
+        else:
+            if deploy_config.get("check_drone_start_pos", True):
+                real_pos, real_quat = self._query_drone_pose()
+                nominal_pos = np.array(self.drones.pos[self.rank])
+                check_drone_start_pos(nominal_pos=nominal_pos, real_pos=real_pos, rng_config=self.randomizations, drone_name=self.drone_name)
```

Comment:
I'm happy to sacrifice a few ms for the sanity check. Also, it's nice to reduce logic complexity.

```suggestion
        if deploy_config.get("check_drone_start_pos", True):
            real_pos, real_quat = self._query_drone_pose()
            nominal_pos = np.array(self.drones.pos[self.rank])
            check_drone_start_pos(nominal_pos=nominal_pos, real_pos=real_pos, rng_config=self.randomizations, drone_name=self.drone_name)
```